packages/core/entity/FileDetail/index.ts

```diff
@@ -135,34 +194,18 @@ export default class FileDetail {
     }
 
     public get localPath(): string | null {
-        // REMOVE THIS (BACKWARDS COMPAT)
-        if (this.path.startsWith("/allen")) {
-            return this.path;
+        if (this.getAnnotation("Cache Eviction Date")) {
+            return FileDetail.generateFilePath(this.env, this.name, this.id);
         }
-
-        const localPath = this.getFirstAnnotationValue("Local File Path");
-        if (localPath === undefined) {
-            return null;
-        }
-        return localPath as string;
+        return null;
     }
 
     public get cloudPath(): string {
-        // REMOVE THIS (BACKWARDS COMPAT)
-        if (this.path.startsWith("/allen")) {
```

Comment:
We still need backward compatibility - should this change (and the block below) use the new heuristic `this.getAnnotation("Cache Eviction Date")`?

Reply:
The backwards compatibility is to handle file.path being '/allen' or the updated version where file.path = cloud path. I thought that we were fully past that after the new file rebuild, hence trying to remove it. however seems like there's still a possibility of file.path being /allen as talked about today in #cloud-first-fms so I am in favor of adding it back

Reply:
In regards to `Cache Eviction Date` I don't think that we should use this yet since it has not been backfilled and will only show up on new annotations. We would not reach a state where the file path property was old where we had an eviction date I don't think?

Reply:
Resolved in [03c770b](https://github.com/AllenInstitute/biofile-finder/pull/389/commits/03c770bc20447a5d27bbb7dde49360e254854308)